src/pages/en/guides/rss.md

```diff
@@ -6,11 +6,143 @@ description: An intro to RSS in Astro
+
+Now, import the `rss` helper from the `@astrojs/rss` package and call with the following parameters:
+
+```js
+// src/pages/rss.xml.js
+import rss from '@astrojs/rss';
+
+export const get = () => rss({
+    // `<title>` field in output xml
+    title: 'Buzz’s Blog',
+    // `<description>` field in output xml
+    description: 'A humble Astronaut’s guide to the stars',
+    // list of `<item>`s in output xml
+    // simple example: generate items for every md file in /src/pages
+    // see "Generating items" section for required frontmatter and advanced use cases
+    items: import.meta.glob('./**/*.md'),
+    // (optional) inject custom xml
+    customData: `<language>en-us</language>`,
+  });
+```
+
+### Generating `items`
+
+The `items` field accepts either:
+1. [An `import.meta.glob(...)` result](#2-importmetaglob-result) **(only use this for `.md` files within the `src/pages/` directory!)**
+2. [A list of RSS feed objects](#1-list-of-rss-feed-objects), each with a `link`, `title`, `pubDate`, and optional `description` and `customData` fields.
+
+#### 1. `import.meta.glob` result
+
+We recommend this option as a convenient shorthand for `.md` files under `src/pages/`. Each post should have a `title`, `pubDate`, and optional `description` and `customData` fields in its frontmatter. If this isn't possible, or you'd prefer to generate this frontmatter in code, [see option 2](#2-list-of-rss-feed-objects).
+
+Say your blog posts are stored under the `src/pages/blog/` directory. You can generate an RSS feed like so:
+
+```js
+// src/pages/rss.xml.js
+import rss from '@astrojs/rss';
+
+export const get = () => rss({
+    title: 'Buzz’s Blog',
+    description: 'A humble Astronaut’s guide to the stars',
+    items: import.meta.glob('./blog/**/*.md'),
+  });
+```
+
+See [Vite's glob import documentation](https://vitejs.dev/guide/features.html#glob-import) for more on this import syntax.
+
+#### 2. List of RSS feed objects
+
+We recommend this option for `.md` files outside of the `pages` directory. This is common when generating routes [via `getStaticPaths`](/en/reference/api-reference/#getstaticpaths). 
+
+For instance, say your `.md` posts are stored under a `src/posts/` directory. Each post has a `title`, `pubDate`, and `slug` in its frontmatter, where `slug` corresponds to the output URL on your site. We can generate an RSS feed using [Vite's `import.meta.globEager` helper](https://vitejs.dev/guide/features.html#glob-import) like so:
+
+```js
+// src/pages/rss.xml.js
+import rss from '@astrojs/rss';
+
+const postImportResult = import.meta.globEager('../posts/**/*.md');
+const posts = Object.values(postImportResult);
+
+export const get = () => rss({
+    title: 'Buzz’s Blog',
+    description: 'A humble Astronaut’s guide to the stars',
+    items: posts.map((post) => ({
+      link: post.frontmatter.slug,
+      title: post.frontmatter.title,
+      pubDate: post.frontmatter.pubDate,
+    }))
+  });
+```
+
+### Adding a stylesheet
```

Comment:
Reading through these docs, I had a thought: It's odd that we have to spend so much time documenting these two different multi-step ways to do something so simple. It's also a bit odd that we generate a static `stylesheet` dynamically like this.

As I user, I realize that what I really want is `stylesheet: '/rss/styles.xsl'`. Instead of vendoring this stylesheet inside the package and making users create two different page routes, why don't we just design for this `public/` workflow only. We could say the following:

--- 

You can style your RSS feed for a more pleasant user experience when viewing the file in your browser.

Use the `rss` function's `stylesheet` option to specify an absolute path to your stylesheet.


```js
rss({
  // ex. use your stylesheet from "public/rss/styles.xsl"
  stylesheet: '/rss/styles.xsl',
  // ...
});
```

If you don't have an RSS stylesheet in mind, we recommend the [Pretty Feed v3](https://github.com/genmon/aboutfeeds/blob/main/tools/pretty-feed-v3.xsl) default stylesheet, which you can download from GitHub and save into your project `public/` directory.

Reply:
Hm okay, would you want to remove `getStylesheet` + the README entry then?